instrumentation/servlet/request-2.3/src/main/java/io/opentelemetry/auto/instrumentation/servlet/v2_3/Servlet2Advice.java

```diff
@@ -57,21 +57,21 @@ public static void stopSpan(
       @Advice.Argument(0) final ServletRequest request,
       @Advice.Argument(1) final ServletResponse response,
       @Advice.Thrown final Throwable throwable,
-      @Advice.Local("otelSpan") Span span,
-      @Advice.Local("otelScope") Scope scope) {
+      @Advice.Local("otelSpan") final Span span,
+      @Advice.Local("otelScope") final Scope scope) {
     if (scope == null) {
       return;
     }
     scope.close();
 
     if (request instanceof HttpServletRequest && response instanceof HttpServletResponse) {
-      TRACER.setPrincipal((HttpServletRequest) request);
+      TRACER.setPrincipal(span, (HttpServletRequest) request);
```

Comment:
span can be `null` here

Reply:
I think `scope != null` (above) implies `span != null` (btw this comment motivated #499)

Reply:
Here it may be true, then `if(span == null)` below should be removed.